internal/util/telemetry/telemetry.go

```diff
@@ -0,0 +1,212 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package telemetry
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"runtime"
+	"time"
+
+	"github.com/AlekSi/pointer"
+	"go.uber.org/zap"
+
+	"github.com/FerretDB/FerretDB/internal/util/state"
+	"github.com/FerretDB/FerretDB/internal/util/version"
+)
+
+// request represents telemetry request.
+type request struct {
+	Version string `json:"version"`
+	Commit  string `json:"commit"`
+	Branch  string `json:"branch"`
+	Dirty   bool   `json:"dirty"`
+	Debug   bool   `json:"debug"`
+	OS      string `json:"os"`
+	Arch    string `json:"arch"`
+
+	UUID   string        `json:"uuid"`
+	Uptime time.Duration `json:"uptime"`
+}
+
+// response represents telemetry response.
+type response struct {
+	LatestVersion string `json:"latest_version"`
+}
+
+// Reporter sends telemetry reports if telemetry is enabled.
+type Reporter struct {
+	*NewReporterOpts
+	c *http.Client
+}
+
+// NewReporterOpts represents reporter options.
+type NewReporterOpts struct {
+	URL            string
+	P              *state.Provider
+	L              *zap.Logger
+	UndecidedDelay time.Duration
+	ReportInterval time.Duration
+	ReportTimeout  time.Duration
+}
+
+// NewReporter create a new reporter.
+func NewReporter(opts *NewReporterOpts) (*Reporter, error) {
```

Comment:
Do we really need `error` in the signature? Do you think we might need it later?

Reply:
I think the next PR will need it. If not, I will remove it